delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/routes/ArchiveRoutes.scala

```diff
@@ -87,6 +93,11 @@ class ArchiveRoutes(
 
   private def sourceToFileResponse(source: AkkaSource): FileResponse =
     FileResponse("archive.tar", MediaTypes.`application/x-tar`, 0L, source)
+
+  private def metadataResponse: Directive1[Boolean] =
```

Comment:
you don't need this, you just need:

```scala
extractRequest.map(req => HeadersUtils.findFirst(req.headers, mediaTypes).isDefined)
```

Reply:
I don't see how that's better, but alright...

Reply:
It is better because we keep using the same methods for the same things concepts.

If you don't find the current method in `HeadersUtils` a useful one or it can be improved, please go on and change it (also changing it where is used, DeltaDirectives and so on).

Something like that might be better than what we currently have:

```
def findFirst(accept: Accept, serviceMediaTypes: Seq[MediaType]): Option[MediaType]= ???
```

Then we benefit everywhere (not just on the ArchivesRoutes)

Reply:
The accept header may not be present, in which case it's better to just pass along the header collection because it makes it simple at the call site.

But for this function to work with archives, we need to choose whether we want an exact match or not. I've modified it to do just that.

Reply:
you don't need the exact match. Just have a:

```scala
private def metadataResponse: Directive1[Boolean] =
extractRequest.map(req => !HeadersUtils.findFirst(req.headers, Set(`application/x-tar`)).isDefined)
```